File: src/main/java/flux/app/api/session/AuthService.java
package flux.app.api.session;

import flux.app.api.models.UserModel;
import flux.app.api.repositories.UserRepo;
import reactor.core.publisher.Mono;

import org.apache.commons.lang3.RandomStringUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AuthService {
  private final UserRepo userRepo;

  @Autowired
  public AuthService(UserRepo userRepo) {
    this.userRepo = userRepo;
  }

  Mono<UserModel> call() {
    return userRepo.save(
        new UserModel(
            null, RandomStringUtils.random(12, true, true), null,
            null, null, null));
  }
}

File: src/main/kotlin/app/flux/api/domain/session/AuthService.kt
package app.flux.api.domain.session

import app.flux.api.models.UserModel
import app.flux.api.repositories.UserRepo
import io.jsonwebtoken.Jwts
import org.apache.commons.lang3.RandomStringUtils
import org.springframework.stereotype.Service

@Service
class AuthService(private val authProperties: AuthProperties, private val userRepo: UserRepo) {
  fun call(): String {
    val user = userRepo.save(UserModel(login = RandomStringUtils.random(12, true, true)))

    return generateToken(user)
  }

  private fun generateToken(user: UserModel): String {
    return Jwts.builder()
      .signWith(authProperties.privateKey)
      .setSubject(user.id.toString())
      .compact()
  }
}
